### PR TITLE
[fix] mc_logits.shape doesn't match mc_labels.shape

### PR DESCRIPTION
**Pull request by flybird11111**
# What does this PR do?

when 'loss_fct(mc_logits.view(-1, mc_logits.size(-1)), mc_labels.view(-1))' coompute the mc_loss, it will raise a error that the logits's shape doesn't match the labels' shape, while the logits and labels are the inputs of loss_fcn.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by amyeroberts** [comment]
Hi @flybird11111, thanks for opening a PR! Could you write in the description what issue this is resolving?

**Comment by flybird11111** [comment]
> Hi @flybird11111, thanks for opening a PR! Could you write in the description what issue this is resolving?

I have described the prolem I encountered in the description.

**Comment by flybird11111** [comment]
> # What does this PR do?
> when 'loss_fct(mc_logits.view(-1, mc_logits.size(-1)), mc_labels.view(-1))' coompute the mc_loss, it will raise a error that the logits's shape doesn't match the labels' shape when batch > 1, while the logits and labels are the inputs of loss_fcn.
> 
> Fixes # (issue)
> 
> ## Before submitting
> * [ ]  This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
> * [ ]  Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
>   Pull Request section?
> * [ ]  Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
>   to it if that's the case.
> * [ ]  Did you make sure to update the documentation with your changes? Here are the
>   [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
>   [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
> * [ ]  Did you write any new necessary tests?
> 
> ## Who can review?
> Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.
